core/src/main/java/org/jdbi/v3/core/mapper/EnumMapperFactory.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jdbi.v3.core.mapper;
+
+import java.lang.reflect.Type;
+import java.util.Optional;
+import org.jdbi.v3.core.config.ConfigRegistry;
+
+/**
+ * Column mapper factory which knows how to map {@link Enum} instances.
+ */
+class EnumMapperFactory implements ColumnMapperFactory {
+    private static final EnumByNameMapperFactory BY_NAME = new EnumByNameMapperFactory();
```

Comment:
Again--why not just register the existing `EnumByNameMapperFactory` directly instead of adding this factory? That class is already public API.

If the intent is to be able to swap out the default enum binding / mapping strategy, let's save that for a separate PR.

Reply:
Just noticed this class is now non-public--so this comment doesn't need to block merge.

Reply:
Well if we just register one or the other, we wouldn't be able to switch the implementation on the fly when the user configures it, we'd need to register the other on top, forever filling up the factories list. I think this should stay as it is for now, it's a necessary abstraction.